### ISSUE TITLE
ERROR: Unable to locate element: {"method":"css selector","selector":"[aria-label="Expand Your Library"]"} (also occurs with "Collapse Your Library")

### ISSUE THREAD

**Issue by MichaelMoldoye** [Kan1shak/spotify-to-ytm#9]
Hi There,

PREAMBLE: 
Had no issues cloning repo and installing dependencies (requirements.txt). I tried using Edge (since Selenium-based, like Chrome / Chromium), but that didn't work. Eventually, I was able run this program successfully just once, using a Chromium browser downloaded via Chocolatey, which still failed over & over until I figured out how to manually edit my Python's Selenium Webdrivers Chromium Options.py script to designate this new app's "chrome.exe" file location. 

PRESENT ISSUE: 
Now, running gui.py per the readme instructions, the localhost:5001 page doesn't load my Spotify library after processing the 2 logins in the new 2nd window. The issue (or, at least the outcome) remains the same while running the program under various test conditions, such as retaining AND clearing/resetting Chromium's login cookies for the 2 webpages, setting Spotify account language from "English (English)" to "English (United Kingdom)" and BACK, etc. 

I continue to encounter the following error(s), seemingly related to (not) recognizing the (changing?) name of the web element/button on Spotify's left-hand-side library, different depending on whether it was previously left expanded or collapsed. 

Any advice for overcoming this, or, additional information required from me to support troubleshooting? Thanks. 
 
`````
Link: http://localhost:5001
INFO:     Will watch for changes in these directories: ['C:\\Users\\*\\github-repos\\spotify-to-ytm']
INFO:     Uvicorn running on http://0.0.0.0:5001 (Press CTRL+C to quit)
INFO:     Started reloader process [25688] using WatchFiles
INFO:     Started server process [19460]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     127.0.0.1:63712 - "GET / HTTP/1.1" 200 OK
INFO:     127.0.0.1:63713 - "GET /start?start-button= HTTP/1.1" 307 Temporary Redirect
INFO:     127.0.0.1:63713 - "GET /check_auth HTTP/1.1" 200 OK

[...]

INFO:     127.0.0.1:63713 - "GET /check_auth HTTP/1.1" 200 OK
INFO:     127.0.0.1:64274 - "GET /update_login?status=false&type=spotify HTTP/1.1" 200 OK
INFO:     127.0.0.1:63713 - "GET /check_auth HTTP/1.1" 200 OK
INFO:     127.0.0.1:64326 - "GET /check_user_confirmation HTTP/1.1" 200 OK

[...]

INFO:     127.0.0.1:50879 - "GET /check_user_confirmation HTTP/1.1" 200 OK
INFO:     127.0.0.1:50900 - "GET /user_confirm_login?done-button= HTTP/1.1" 200 OK
INFO:     127.0.0.1:50909 - "GET /check_user_confirmation HTTP/1.1" 200 OK
INFO:     127.0.0.1:50900 - "GET /is_library_built HTTP/1.1" 200 OK
INFO:     127.0.0.1:50900 - "GET /is_library_built HTTP/1.1" 200 OK
INFO:     127.0.0.1:50900 - "GET /is_library_built HTTP/1.1" 200 OK
INFO:     127.0.0.1:50900 - "GET /is_library_built HTTP/1.1" 200 OK
INFO:     127.0.0.1:50942 - "GET /update_login?status=true HTTP/1.1" 200 OK
INFO:     127.0.0.1:50900 - "GET /is_library_built HTTP/1.1" 200 OK
INFO:     127.0.0.1:50900 - "GET /is_library_built HTTP/1.1" 200 OK
INFO:     127.0.0.1:50900 - "GET /is_library_built HTTP/1.1" 200 OK
INFO:     127.0.0.1:50900 - "GET /is_library_built HTTP/1.1" 200 OK
INFO:     127.0.0.1:50900 - "GET /is_library_built HTTP/1.1" 200 OK
INFO:     127.0.0.1:50900 - "GET /is_library_built HTTP/1.1" 200 OK
INFO:     127.0.0.1:50900 - "GET /is_library_built HTTP/1.1" 200 OK
INFO:     127.0.0.1:50900 - "GET /is_library_built HTTP/1.1" 200 OK
INFO:     127.0.0.1:50900 - "GET /is_library_built HTTP/1.1" 200 OK
INFO:     127.0.0.1:50900 - "GET /is_library_built HTTP/1.1" 200 OK
INFO:     127.0.0.1:50900 - "GET /is_library_built HTTP/1.1" 200 OK
INFO:     127.0.0.1:50900 - "GET /is_library_built HTTP/1.1" 200 OK
INFO:     127.0.0.1:50900 - "GET /is_library_built HTTP/1.1" 200 OK
Exception in thread Thread-1 (start_sess):
Traceback (most recent call last):
  File "C:\Users\*\github-repos\spotify-to-ytm\src\setup.py", line 101, in _get_library_auth
    WebDriverWait(self.driver, 5).until(
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        EC.presence_of_element_located((By.CSS_SELECTOR, '[aria-label="Expand Your Library"]'))
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "C:\Users\*\AppData\Local\Programs\Python\Python313\Lib\site-packages\selenium\webdriver\support\wait.py", line 146, in until
    raise TimeoutException(message, screen, stacktrace)
selenium.common.exceptions.TimeoutException: Message:
Stacktrace:
        GetHandleVerifier [0x00AAC7F3+24435]
        (No symbol) [0x00A32074]
        (No symbol) [0x009006E3]
        (No symbol) [0x00948B39]
        (No symbol) [0x00948E8B]
        (No symbol) [0x00991AC2]
        (No symbol) [0x0096D804]
        (No symbol) [0x0098F20A]
        (No symbol) [0x0096D5B6]
        (No symbol) [0x0093C54F]
        (No symbol) [0x0093D894]
        GetHandleVerifier [0x00DB70A3+3213347]
        GetHandleVerifier [0x00DCB0C9+3295305]
        GetHandleVerifier [0x00DC558C+3271948]
        GetHandleVerifier [0x00B47360+658144]
        (No symbol) [0x00A3B27D]
        (No symbol) [0x00A38208]
        (No symbol) [0x00A383A9]
        (No symbol) [0x00A2AAC0]
        BaseThreadInitThunk [0x757A7BA9+25]
        RtlInitializeExceptionChain [0x774AC2EB+107]
        RtlClearBits [0x774AC26F+191]


During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\*\AppData\Local\Programs\Python\Python313\Lib\threading.py", line 1041, in _bootstrap_inner
    self.run()
    ~~~~~~~~^^
  File "C:\Users\*\AppData\Local\Programs\Python\Python313\Lib\threading.py", line 992, in run
    self._target(*self._args, **self._kwargs)
    ~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\*\github-repos\spotify-to-ytm\gui.py", line 23, in start_sess
    spot = SpotifyManager()
  File "C:\Users\*\github-repos\spotify-to-ytm\src\spotify.py", line 23, in __init__
    self.client_token, self.authorization = self.session.get_library()
                                            ~~~~~~~~~~~~~~~~~~~~~~~~^^
  File "C:\Users\*\github-repos\spotify-to-ytm\src\setup.py", line 128, in get_library
    client_token, authorization, persisted_query = self._get_library_auth()
                                                   ~~~~~~~~~~~~~~~~~~~~~~^^
  File "C:\Users\*\github-repos\spotify-to-ytm\src\setup.py", line 108, in _get_library_auth
    library_button = self.driver.find_element(By.CSS_SELECTOR,'[aria-label="Expand Your Library"]')
  File "C:\Users\*\AppData\Local\Programs\Python\Python313\Lib\site-packages\selenium\webdriver\remote\webdriver.py", line 898, in find_element
    return self.execute(Command.FIND_ELEMENT, {"using": by, "value": value})["value"]
           ~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\*\AppData\Local\Programs\Python\Python313\Lib\site-packages\selenium\webdriver\remote\webdriver.py", line 429, in execute
    self.error_handler.check_response(response)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^
  File "C:\Users\*\AppData\Local\Programs\Python\Python313\Lib\site-packages\selenium\webdriver\remote\errorhandler.py", line 232, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {"method":"css selector","selector":"[aria-label="Expand Your Library"]"}
  (Session info: chrome=134.0.6998.178); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception
Stacktrace:
        GetHandleVerifier [0x00AAC7F3+24435]
        (No symbol) [0x00A32074]
        (No symbol) [0x009006E3]
        (No symbol) [0x00948B39]
        (No symbol) [0x00948E8B]
        (No symbol) [0x00991AC2]
        (No symbol) [0x0096D804]
        (No symbol) [0x0098F20A]
        (No symbol) [0x0096D5B6]
        (No symbol) [0x0093C54F]
        (No symbol) [0x0093D894]
        GetHandleVerifier [0x00DB70A3+3213347]
        GetHandleVerifier [0x00DCB0C9+3295305]
        GetHandleVerifier [0x00DC558C+3271948]
        GetHandleVerifier [0x00B47360+658144]
        (No symbol) [0x00A3B27D]
        (No symbol) [0x00A38208]
        (No symbol) [0x00A383A9]
        (No symbol) [0x00A2AAC0]
        BaseThreadInitThunk [0x757A7BA9+25]
        RtlInitializeExceptionChain [0x774AC2EB+107]
        RtlClearBits [0x774AC26F+191]

INFO:     127.0.0.1:50900 - "GET /is_library_built HTTP/1.1" 200 OK

**Comment by lukas-analyst** [comment]
In the **setup.py** (file) locate text "_Expand_" and replace it for "_Open_". Then it will work nomally. 

You need your Spotify app to be in English (click on your photo-icon and go to settings --> language --> English)

**Comment by Kan1shak** [comment]
Hey @lukas-analyst @MichaelMoldoye , thanks for letting me know! The fix has been added in the latest commit.
